### PR TITLE
fix(reg): Fix wasm external components are not hit testable

### PR DESCRIPTION
**Pull request by dr1rrb**
fixes https://github.com/unoplatform/uno/issues/7848

## Bugfix
Regression: [WASM] External components does not allow pointers interaction

## What is the current behavior?
Since https://github.com/unoplatform/uno/pull/7571, setting the `Background` on `FrameworkElement` is no longer sufficient for an element to be hit testable. This breaks HTML external components which was (wrongly) relying on this to be hit testable (as stated in doc [here]( https://github.com/unoplatform/uno/blob/f8bd42650d69a41b4360f3e533e21c7e789984ae/doc/articles/interop/wasm-javascript-3.md?plain=1#L105)) and has no other way to enable pointers.

## What is the new behavior?
WASM external component are now always considered as hit testable.

## PR Checklist
- [x] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [x] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] ~~Contains **NO** breaking changes~~, cf below
- [x] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [x] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information
```
                                  *******************
                                  * BREAKING CHANGE *
                                  *******************
```

This was actually introduced by https://github.com/unoplatform/uno/pull/7571

Setting the `Background` of a `FrameworkElement` is no longer a valid way to force element to be hit testable. You can restore previous behavior by setting config flag `Uno.UI.FeatureConfiguration.FrameworkElement.UseLegacyHitTest` to `true`

### CODE REVIEW

File: src/Uno.UI/UI/Xaml/HtmlElementHelper.wasm.cs [hunk 1]
```diff
@@ -0,0 +1,70 @@
+﻿using System;
```
Comment (jeromelaban) [inline src/Uno.UI/UI/Xaml/HtmlElementHelper.wasm.cs:1]:
```suggestion
#nullable enable
﻿using System;
```

File: src/Uno.UI/UI/Xaml/UIElement.wasm.cs [hunk 2]
```diff
@@ -176,6 +135,8 @@ private string GetHtmlTag(string htmlTag)
 
 		public bool HtmlTagIsSvg { get; }
 
+		internal bool HtmlTagIsExternallyDefined { get; }
```
Comment (jeromelaban) [inline src/Uno.UI/UI/Xaml/UIElement.wasm.cs:138]: We need flags for these.

**Review by davidjohnoliver — APPROVED** [review]

**Review by MartinZikmund — APPROVED** [review]

**Pull request merged**